modules/core/src/test/java/org/apache/ignite/internal/util/ByteUtilsTest.java

```diff
@@ -43,4 +48,19 @@ void testStringFromBytes() {
         assertEquals("", stringFromBytes(stringToBytes("")));
         assertEquals("abc", stringFromBytes(stringToBytes("abc")));
     }
+
+    @Test
+    void testComparableLong() {
```

Comment:
The method is pretty subtle, so we should probably have more (separate) tests. I would suggest to have the following:

1. A test that makes sure that forward-then-backward conversion yields the same result we had in the beginning (this is checked by the first line of the test)
2. For each of 8 bytes, a test that makes sure that, if just Nth byte differs, and in both cases the value of the byte has the same sign, the order is as expected
3. For each of 8 bytes, a test that makes sure that, if just Nth byte differs, and the values of the byte has different signs, the order is as expected

Parameterization could be used for 2 and 3.

Reply:
Isn't that an overkill? It's really hard to implement this method wrongly

Reply:
I also have check where different bytes differ (1 vs 256), it validates that we use BE

Reply:
There is a constant with 8 and 0 intermingled. It's easy to mess them up in this pattern, and not so easy to spot...